ApollosProject/apollos-apps
# fix: remove video select

### **Purpose**
The purpose of this PR is to ensure the video select feature only works in the case where videos have names. Conrad mentions in the relevant basecamp post [here](https://3.basecamp.com/3926363/buckets/27724884/todos/5175263026#:~:text=video%20select%20was%20intended%20to%20select%20between%20different%20types%20of%20videos%20as%20in%20%2D%20English%2C%20Spanish%2C%20or%20Modern%20Service%20and%20Traditional%20Service) (which I recommend looking at as well), that

> video select was intended to select between different types of videos as in - English, Spanish, or Modern Service and Traditional Service

whereas for some clients, specifically Crossings, video select is picking up 3 videos

<img width="910" alt="Screen Shot 2022-08-17 at 10 53 19" src="https://user-images.githubusercontent.com/19194337/185171682-3f66f2cd-1cb9-43af-984d-6d9194950e43.png">

which correspond to these in rock:

<img width="815" alt="Screen Shot 2022-08-17 at 10 55 08" src="https://user-images.githubusercontent.com/19194337/185172184-03706aab-7a9f-468c-81fd-d5afb73fd502.png">


### **Solution**
**Since video select is picking up videos that don't match its intended purpose and since we are not handling null names, it seems appropriate to use the lack of a name as a way to disable videoselect when and if it is not used as intended, without disabling it entirely.**

_*NOTE: This solution below is based on the assumption that when used as intended (i.e. with specific language options or service types), names wont be null and the video select feature works. The solution I propose above also tries to avoid disabling the component entirely in case other churches are using video select with the intended types. Shoot me down here if I'm not making sense or if I need to dig deeper. Ultimately I'm looking for a way to disable videoselect for cases where it's not being passed the right type of videos.*_


### **Testing**
**Setup**
1. Clone the repo and checkout the following branch: `handle-video-select-null-names`
2. Pick any multi-tenant app (I'm using crossings_community_church) for a fresh spin up and run the setup script (`setup-app.sh`)
3. Run the iOS or Android app

**Steps to Test**
1. Login
2. Select any video item
3. You should no longer see the video select button if _any_ video.name in the videos array is null

### **Before**
![Screen Shot 2022-08-17 at 12 27 18](https://user-images.githubusercontent.com/19194337/185192938-07ae3f61-8a79-4ddd-b2d4-84da7a9b66c2.png)


### **After**
![Screen Shot 2022-08-17 at 12 28 01](https://user-images.githubusercontent.com/19194337/185193064-4ab559cc-3c8b-467b-83b3-5c6a42ce568e.png)
